### PR TITLE
[Hack Week] Post success view

### PR DESCRIPTION
**Pull request by kean**
Fixes #https://github.com/wordpress-mobile/WordPress-iOS/issues/21509.

I've been going through the maintenance board, noticed the linked defect, and decided to resurrect this PR from the last Hack Week, which I didn't get to complete then. The new screen looks more like what you see in Gutenberg, and has a new "Promote with Blaze" button.

The initial idea was to show this automatically as part of the Prepublishing sheet, but this is a more conservative approach that simply replaced the current screen (with a slightly broken layout).

The design comes from one of the prototypes made with Chris, but with some minor changes. There've been so many versions of this, I don't any particular direction was picked as the final one in the end.

## To test:

- Publish a new post
- Tap "View" on a snackbar
- ✅ Verify that the updated success screen is shown
- ✅ Verify that all buttons are working
- ✅ Verify that for Blaze-enabled blogs, "Promote with Blaze" button is visible.

## Regression Notes
1. Potential unintended areas of impact: Post Publish Success Notice View
2. What I did to test those areas of impact (or what existing automated tests I relied on): manual
3. What automated tests I added (or what prevented me from doing so): n/a

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Podfile.lock [hunk 1]
```diff
@@ -197,7 +197,7 @@ SPEC CHECKSUMS:
   Gifu: 416d4e38c4c2fed012f019e0a1d3ffcb58e5b842
   Gravatar: 51437de6811c1d8d6f60c52985f2ca00a85cfc8e
   Gridicons: 4455b9f366960121430e45997e32112ae49ffe1d
-  Gutenberg: 3117b6fe578fb7f0bb75377ba96f436ee05c30fe
+  Gutenberg: cd6259542b7078fe875d51b21c962d02bac6c032
```
Comment (kean) [inline Podfile.lock]: Not sure why it was added – will investigate tomorrow.  When I run `pod install` in `trunk` it also changes the Podfile.lock.

File: WordPress/Classes/Utility/Analytics/WPAnalyticsEvent.swift [hunk 2]
```diff
@@ -1595,7 +1595,7 @@ import Foundation
         case .assertionFailure:
             return "assertion_failure"
         case .postCoordinatorErrorEncountered:
-            return "post-coordinator-eerror-encountered"
+            return "post-coordinator-error-encountered"
```
Comment (kean) [inline WordPress/Classes/Utility/Analytics/WPAnalyticsEvent.swift]: Found this typo during testing.

**Review by momo-ozawa — COMMENTED** [review]
It works as described, but I have some design questions

* Would it make more sense to group the CTAs together? (i.e. Share, Promote with Blaze, Done are all full width buttons vertically stacked)
* Is there a way we can avoid truncating "View on url..."?

File: WordPress/Resources/AppImages.xcassets/post-published.imageset/post-publihsed.svg
Comment (momo-ozawa) [inline WordPress/Resources/AppImages.xcassets/post-published.imageset/post-publihsed.svg:1]: [nit] typo in filename

**Comment by kean** [comment]
> Would it make more sense to group the CTAs together? (i.e. Share, Promote with Blaze, Done are all full width buttons vertically stacked)

Yeah, there've been multiple options of this design. This is something that wasn't tried: using a grouped list, which I think works best if you need to stack multiple buttons vertically. So perhaps this will work better?

Screenshot 2024-05-01 at 7 27 43 AM

**Comment by kean** [comment]
Hey, I updated the design based on what we discussed.

Screenshot 2024-05-01 at 11 23 42 AM

**Review by momo-ozawa — APPROVED** [review]
LGTM! Did you figure out the Podfile.lock issue?

**Comment by kean** [comment]
> LGTM! Did you figure out the Podfile.lock issue?

Looks like it got updated in `trunk`. The change disappear after the rebase.

File: WordPress/Resources/AppImages.xcassets/post-published.imageset/post-publihsed.svg
Comment (kean) [reply WordPress/Resources/AppImages.xcassets/post-published.imageset/post-publihsed.svg:1]: Fixed

**Pull request merged**